lib/matplotlib/axes/_secondary_axes.py

```diff
@@ -306,7 +306,7 @@ def set_xlabel(self, xlabel, fontdict=None, labelpad=None, **kwargs):
         xlabel : str
             The label text.
 
-        labelpad : scalar, optional, default: None
+        labelpad : float, default: ``self.xaxis.labelpad``
```

Comment:
I think we need to decide how to document the "None means something" pattern which occurs quite frequently throughout the codebase.  Do we mention explicitly the ability to pass None ("float or None"), and do we say "default: None" and later explain in the text that None means something something?
I think your approach (don't mention None and directly give the "effective" default) is much more practical, but this should be codified to avoid going back and forth with edits...

Reply:
I would only document `default: None` explicitly if it has a dedicated meaning in the sense that it makes sense to pass `func(param=None)`. In cases that just use None as a sentinel value for "not passed" I would not mention that in the docs. In these cases the sentinel value is not of interest to the user.

Regardless of that it's API and cannot be changed, but we don't have to advertise it in the semantic description of the docstring.

Reply:
Actually I think I have somewhere a patch for making a decorator that deprecates these "None as default" in the cases where the default can just as easily be spelled out (typically 0 or 1 or the empty string)... anyways that's a plan for another time.